tests/test_utilities.py

```diff
@@ -227,3 +229,47 @@ def test_get_course_jobs(self):
             assert job_skill.skill == course_skills.skill
+
+    @mock.patch('taxonomy.utils.translate_text')
+    def test_get_or_create_course_full_description_translation_success(self, translate_text_mocked):
+        """
+        Validate that `get_or_create_course_full_description_translation` returns translated course_description
+         and creates Translation object.
+        """
+        course_description = "abc def"
+        translated_course_description = "different text"
+        translate_text_mocked.return_value = {'SourceLanguageCode': ENGLISH,
+                                              'TranslatedText': translated_course_description
+                                              }
+        factories.TranslationFactory(source_record_identifier=COURSE_KEY,
+                                     source_model_field='full_description',
+                                     source_model_name='Course',
+                                     source_text=course_description,
+                                     translated_text=translated_course_description,
+                                     translated_text_language=ENGLISH,
+                                     source_language=ENGLISH,
+                                     )
+
+        expected_translated_description = utils.get_or_create_course_full_description_translation(COURSE_KEY,
+                                                                                                  course_description)
+        translation_record = Translation.objects.filter(source_model_name='Course',
+                                                        source_model_field='full_description',
+                                                        source_record_identifier=COURSE_KEY).exists()
+        assert translation_record is True
+        assert translated_course_description == expected_translated_description
+
+    @mock.patch('taxonomy.utils.translate_text')
+    def test_get_or_create_course_full_description_translation_error(self, translate_text_mocked):
+        """
+        Validate that `get_or_create_course_full_description_translation` returns actual course_description
+         if translate_text method returns None and does not create Translation object.
+        """
+        course_description = "abc def"
+        translate_text_mocked.return_value = None
+        expected_translated_description = utils.get_or_create_course_full_description_translation(COURSE_KEY,
+                                                                                                  course_description)
+        translation_record = Translation.objects.filter(source_model_name='Course',
+                                                        source_model_field='full_description',
+                                                        source_record_identifier=COURSE_KEY).first()
+        assert translation_record is None
+        assert course_description == expected_translated_description
```

Comment:
can we test the remaining scenarios as well? I think there should be six of them

Reply:
Done